src/customer/strategies/bolt/bolt-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+    }
+
+    continueAsGuest(credentials: GuestCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this.openCustomBoltCheckoutWidget(credentials.email, methodId)
+            .then(() => this._store.dispatch(this._billingAddressActionCreator.continueAsGuest(credentials, options)))
+            .then(() => this._store.getState());
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.signInCustomer(credentials, options))
+            .then(() => this.openCustomBoltCheckoutWidget(credentials.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._customerActionCreator.signOutCustomer(options));
+    }
+
+    signUp(customerAccount: CustomerAccountRequestBody, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.createCustomer(customerAccount, options))
+            .then(() => this.openCustomBoltCheckoutWidget(customerAccount.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    private openCustomBoltCheckoutWidget(email: string, methodId: string): Promise<InternalCheckoutSelectors> {
+        return new Promise(resolve => {
+            const state = this._store.getState();
+            const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+            try {
+                if (!this._boltClient) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                }
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (paymentMethod.initializationData?.customContinueFlowEnabled) {
+                    const callbacks = {
+                        close: resolve,
+                    };
+
+                    this._boltClient.openCheckout(email, callbacks)
+                        .then(result => !result && resolve());
+                } else {
+                    resolve();
+                }
+            } catch (error) {
+                resolve();
+                throw error;
```

Comment:
per our discussion, it worth to just log an error in order to not interrupt the default flow (will check if `throw` fully interrupts the default flow). continueWith strategy is not as vital for checkout as a strategy of a payment provider. we can remain the default flow if we have any issue

Reply:
I checked it and it is working as expected. If we somehow get an error, we will continue with default flow without any problems

Reply:
I think the intention is good. But I think what we should do, from SDK library's perspective, is to return an error and let the client to catch and decide what to do next as a recovery. The client may choose to silently ignore the error and log it behind the scene. Or it may choose to display a message informing that the shopper that something went wrong but they can still progress.